RecoPixelVertexing/PixelTriplets/plugins/GPUCACell.h

```diff
@@ -159,19 +175,23 @@ class GPUCACell {
     tmpNtuplet.push_back_unsafe(theDoubletId);
     assert(tmpNtuplet.size()<=4);
 
-    if(theOuterNeighbors.size()>0) { // continue
+    if(theOuterNeighbors.size()>0) {
       for (int j = 0; j < theOuterNeighbors.size(); ++j) {
         auto otherCell = theOuterNeighbors[j];
-        cells[otherCell].find_ntuplets(cells, foundNtuplets, apc, tmpNtuplet,
-                                       minHitsPerNtuplet);
+        cells[otherCell].find_ntuplets(hh, cells, foundNtuplets, apc, tupleMultiplicity, 
+                                       tmpNtuplet, minHitsPerNtuplet);
       }
     } else {  // if long enough save...
       if ((unsigned int)(tmpNtuplet.size()) >= minHitsPerNtuplet-1) {
-        hindex_type hits[6]; auto nh=0U;
-        for (auto c : tmpNtuplet) hits[nh++] = cells[c].theInnerHitId;
-        hits[nh] = theOuterHitId; 
-        uint16_t it = foundNtuplets.bulkFill(apc,hits,tmpNtuplet.size()+1);
-        for (auto c : tmpNtuplet) cells[c].theTracks.push_back(it);
+        // triplets accepted only pointing to the hole
+        if (tmpNtuplet.size()>=3 || hole(hh, cells[tmpNtuplet[0]])) {
```

Comment:
This additional runtime condition (that for minHitsPerNtuplet==4 is always true and should not require hole to be invoked) seems NOT to be particularly cheap:
   * before
```12.44%  2.99829s     20000  149.91us  26.912us  617.05us  kernel_find_ntuplets(GPUCACell*, unsigned int const *, HistoContainer<unsigned int, unsigned int=10000, unsigned int=50000, unsigned int=32, unsigned short, unsigned int=1>*, AtomicPairCounter*, unsigned int)```
   * after
```12.44%  2.99829s     20000  149.91us  26.912us  617.05us  kernel_find_ntuplets(GPUCACell*, unsigned int const *, HistoContainer<unsigned int, unsigned int=10000, unsigned int=50000, unsigned int=32, unsigned short, unsigned int=1>*, AtomicPairCounter*, unsigned int)```